CRM/Contact/BAO/Contact.php

```diff
@@ -3642,4 +3642,16 @@ public static function isFieldHasLocationType($fieldTitle) {
     return FALSE;
   }
 
+  /**
+   * Checks permission to create new contacts from entityRef widget
+   *
+   * Note: other components must return an array of links from this function,
```

Comment:
1. Looking in 5.8.0 there were no instantiations of this function. Should this read, '...other components will need to eventually...'? 
2. Is your plan to display create links from Common.js when user has permission to create an entity of that type? 
3. Shouldn't the comment indicate that this is possible for any entity that provides a BAO, whether it is for a component or from an extension?

Reply:
That function is a new invention in this PR. It's a callback for any BAO to provide links for creating a new item from within the EntityRef widget. Prior to this PR only contacts could do that and the links were hardcoded to be "New Individual" "New Household" and "New Organization".

This is my attempt to add more flexibility and it will probably evolve a little more but I'm mostly happy with it.